vendor/csv_to_rkg.py

```diff
@@ -0,0 +1,186 @@
+import csv
+import binascii
+from dataclasses import dataclass
+
+
+# fmt: off
+class FaceButtonInput:
+    def __init__(self, AButton: int, BButton: int, ItemButton: int, csv_line: int):
+        assert AButton == 0 or AButton == 1, \
+            f"A button input has value `{AButton}` at line {csv_line}"
+
+        assert BButton == 0 or BButton == 1, \
+            f"B button input has value `{BButton}` at line {csv_line}"
+
+        assert ItemButton == 0 or ItemButton == 1, \
+            f"Item button input has value `{ItemButton}` at line {csv_line}"
+
+        self.AButton = AButton
+        self.BButton = BButton
+        self.ItemButton = ItemButton
+
+    def __eq__(self, other):
+        return (
+            self.AButton == other.AButton
+            and self.BButton == other.BButton
+            and self.ItemButton == other.ItemButton
+        )
+
+
+class DirectionInput:
+    def __init__(self, YDirection: int, XDirection: int, csv_line: int):
+        assert -7 <= YDirection <= 7, \
+            f"Y direction input has value `{YDirection}` at line {csv_line}"
+
+        assert -7 <= XDirection <= 7, \
+            f"X direction input has value `{XDirection}` at line {csv_line}"
+
+        self.YDirection = YDirection + 7
+        self.XDirection = XDirection + 7
+
+    def __eq__(self, other):
+        return (
+            self.YDirection == other.YDirection and self.XDirection == other.XDirection
+        )
+
+
+class TrickInput:
+    def __init__(self, TrickInput: int, csv_line: int):
+        assert 0 <= TrickInput <= 4, \
+            f"Trick input has value `{TrickInput}` at line {csv_line}"
```

Comment:
"Trick input on line {csv_line} is out of range! Expected 0 - 4, got \`{TrickInput}\`"